src/Symfony/Component/Console/Command/Command.php

```diff
@@ -648,6 +654,29 @@ public function getHelper(string $name): HelperInterface
         return $this->helperSet->get($name);
     }
 
+    private function writeDeprecationMessages(InputDefinition $inputDefinition, InputInterface $input, OutputInterface $output): void
+    {
+        $deprecationMessages = [];
+        foreach ($inputDefinition->getOptions() as $inputOption) {
+            if ($inputOption->isDeprecated()) {
+                try {
+                    $optionName = $inputOption->getName();
+                    $optionValue = $input->getOption($optionName);
+                    if (isset($optionValue) && $optionValue !== $inputOption->getDefault()) {
+                        $deprecationMessages[] = sprintf('The option "--%s" is deprecated.', $optionName);
```

Comment:
Non-blocker: if people use the shortcut, this deprecation message mention the full name of the option. They have look at the command definition to find which option have been used.

Reply:
I have included the shortcut in the output, if it exists.